admin/category_product_listing.php

```diff
@@ -25,1230 +25,1656 @@
-          //   multiple observers might be injecting content!
-          // - If heading-columns are added, be sure to add the associated data columns, too, via the
-          //   'NOTIFY_ADMIN_PROD_LISTING_DATA_AFTER_QTY' notification.
-          //
-          if ($show_prod_labels) {
-              $extra_headings = false;
-              $zco_notifier->notify('NOTIFY_ADMIN_PROD_LISTING_HEADERS_AFTER_QTY', '', $extra_headings);
-              if (is_array($extra_headings)) {
-                  foreach ($extra_headings as $heading_info) {
-                      $align = (isset($heading_info['align'])) ? (' text-' . $heading_info['align']) : '';
-?>
-                <th class="hidden-sm hidden-xs<?php echo $align; ?>"><?php echo $heading_info['text']; ?></th>
-<?php
-                  }
-              }
-          }
-?>
-                <th class="text-right"><?php echo TABLE_HEADING_STATUS; ?></th>
-                <?php
-                if ($action === '') {
-                  ?>
-                  <th class="text-right hidden-sm hidden-xs"><?php echo TABLE_HEADING_CATEGORIES_SORT_ORDER; ?></th>
-                  <th class="text-right"><?php echo TABLE_HEADING_ACTION; ?></th>
-                  <?php
+                if (isset($_GET['search'])) {
+                    $keyword_search_fields = [
+                        'cd.categories_name',
+                        'cd.categories_description',
+                    ];
+                    $sql .= zen_build_keyword_where_clause($keyword_search_fields, trim($keywords), true);
+                } else {
+                    $sql .= " WHERE c.parent_id = :category";
+                    $sql = $db->bindVars($sql, ':category', $current_category_id, 'integer');
                 }
+
+                $sql .= $order_by;
+
+                $categories = $db->Execute($sql);
+
+                $show_prod_labels = ($search_result || $categories->EOF);
                 ?>
-              </tr>
-            </thead>
-            <?php
-            foreach ($categories as $category) {
-              $categories_count++;
+                <table id="categories-products-table" class="table table-striped">
+                    <thead>
+                    <tr>
+                        <th class="text-right shrink"><?php
+                            echo TABLE_HEADING_ID; ?></th>
+                        <th><?php
+                            echo TABLE_HEADING_CATEGORIES_PRODUCTS; ?></th>
+                        <th class="hidden-sm hidden-xs"><?php
+                            echo TABLE_HEADING_IMAGE; ?></th>
+                        <?php
+                        if ($show_prod_labels) { ?>
+                            <th class="hidden-sm hidden-xs"><?php
+                                echo TABLE_HEADING_MODEL; ?></th>
+                            <th class="text-right hidden-sm hidden-xs"><?php
+                                echo TABLE_HEADING_PRODUCTS_PRICE; ?></th>
```

Comment:
As above, I find splitting these <th> tags onto multiple lines to less readable.

Reply:
Agreed.